lib/dk-dumpdb/task/internal_task.rb

```diff
@@ -0,0 +1,56 @@
+require 'dk/task'
+require 'much-plugin'
+
+module Dk::Dumpdb;end
+module Dk::Dumpdb::Task
+
+  module InternalTask
+    include MuchPlugin
+
+    plugin_included do
+      include Dk::Task
+      include InstanceMethods
+
+    end
+
+    module InstanceMethods
+
+      private
+
+      def source_cmd!(cmd_str)
+        if params['script'].ssh?
+          ssh!(cmd_str, :hosts => params['script'].ssh)
+        else
+          cmd!(cmd_str)
+        end
+      end
+
+      def target_cmd!(cmd_str)
+        cmd!(cmd_str)
+      end
+
+    end
+
+    module TestHelpers
+      include MuchPlugin
+
+      plugin_included do
+        include Dk::Task::TestHelpers
+        include InstanceMethods
+
+      end
+
+      module InstanceMethods
+
+        def set_dk_dumpdb_script_param(*args, &block)
+          @params ||= {}
+          @params['script'] = Factory.script(*args, &block)
```

Comment:
@kellyredding - Is this intended to be used outside of this gems test suite? If so I'm not sure this will work because the `test/support/factory` is the only factory with the `script` method.

If this is only intended to be an internal helper it might be better for these to live in the `test/support` folder.